### PR TITLE
feat: integrate `DtPresence` within `DtAvatar`

### PR DESCRIPTION
**Pull request by modialpad**
# Enable showing presence in `DtAvatar`

Allow a consumer to set a `presence` prop to show a presence indicator for a particular avatar. 

The position of the presence indicator needs to correspond with the size of the Avatar as outlined here: https://www.figma.com/file/Vw71A2boup3Gb3PqUQTSiu/Avatar-%2B-Presence?node-id=1%3A2&viewport=1966%2C1826%2C3.85&t=v4vpriXLRXUR2F4u-0

## :hammer_and_wrench: Type Of Change

- [x] Feature

## :book: Description

Following the work for adding Presence, the Avatar component will need to be updated to render this new `<Presence/>` component internally whenever a consumer passes a new presence prop.

## :pencil: Checklist

- [x] I have reviewed my changes
- [x] I have added tests
- [x] I have added all relevant documentation
- [x] I have validated components with a screen reader
- [x] I have validated components keyboard navigation
- [x] I have considered the performance impact of my change
- [x] I have checked that my change did not significantly increase bundle size
- [x] I am exporting any new components or constants in the index.js in the component directory
- [x] I am exporting any new components or constants in the index.js in the root

## :camera: Screenshots / GIFs

image

### CODE REVIEW

File: package-lock.json
Comment (modialpad) [inline package-lock.json:22900]: Not sure if all these updates are normal, I bumped DT's version and ran `npm install`. Does npm still upgrade minor versions automatically?

Comment (braddialpad) [reply package-lock.json:22900]: 🤷 npm has a mind of it's own.

File: components/avatar/avatar.vue
Comment (braddialpad) [inline components/avatar/avatar.vue]: What we probably should have done here is include these styles as classes within avatar in the dialtone repo, and then apply these classes based on the avatar size. This would align with our practices for most of our other components, and the classes would be documented in dialtone instead of hidden in the vue component

I'm okay with merging this as is though and then further iterating to make this change.

**Review by braddialpad — APPROVED** [review]
Looks good, just one future change.

File: components/avatar/avatar.vue
Comment (modialpad) [reply components/avatar/avatar.vue]: Np, I think I'm fine with holding this PR until making that change in DT. I've seen that pattern before so I know what you mean - I created a draft [here](https://github.com/dialpad/dialtone/pull/739) based on that. Could you take a look and let me know? I'm not sure if we have a naming standard for targeting DT components within other DT components.

https://github.com/dialpad/dialtone/pull/739

When that's merged we'll just use a constant to map to these classes for presence here.

Comment (francisrupert) [reply components/avatar/avatar.vue]: I'll do a quick PR in `dialtone` to add it to https://dialpad.design/components/avatar.html

Comment (francisrupert) [reply components/avatar/avatar.vue]: @modialpad Sorry I got ahead of it and didn't notice your draft PR. What I did at https://github.com/dialpad/dialtone/pull/740 should do it.

Comment (francisrupert) [reply components/avatar/avatar.vue]: https://github.com/dialpad/dialtone/pull/740 has been merged to staging. I'll defer to Brad on timing of publishing so you can update this PR to use it.

Comment (modialpad) [reply components/avatar/avatar.vue]: Awesome thanks!

Comment (modialpad) [reply components/avatar/avatar.vue]: @braddialpad I updated this w/ @francisrupert's latest changes - let me know if it LG and I'll just merge.

One thing I noticed is that for the presence class modifiers we don't have an explicit modifier for `sm` case. It's actually the default, so it still works as expected, but in the tests I just had to assume that the styles are correct for the default case.

**Pull request merged**
